src/api/githubService.js

```diff
@@ -0,0 +1,286 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import axios from "axios";
+import moment from "moment";
+
+const functionalTestsWorkflowId = 5937682;
+
+const _getSingleWorkflow = async (workflowId) => {
+  const singleWorkflowRunApi = `https://api.github.com/repos/mozilla-mobile/mozilla-vpn-client/actions/workflows/${workflowId}/runs?branch=main`;
+  let return_data = [];
+  let i = 0;
+
+  try {
+    const { data } = await axios.get(singleWorkflowRunApi);
+    while (return_data.length < 12) {
+      const workflow = data.workflow_runs[i];
+      if (
+        workflow.conclusion !== "cancelled" &&
+        workflow.status === "completed"
+      ) {
+        const workflow_run = {
+          id: workflow.id,
```

Comment:
I don't understand why you cannot se 'workflow' directly instead of copying some of the data.